src/python/pants/backend/python/rules/hermetic_pex.py

```diff
@@ -1,34 +1,161 @@
+from pants.util.logging import LogLevel
+from pants.util.memo import memoized_property
+from pants.util.ordered_set import OrderedSet
 from pants.util.strutil import create_path_env_var
 
 
+@dataclass(frozen=True)
+class PexEnvironment(EngineAware):
+    path: Iterable[str]
+    interpreter_search_paths: Iterable[str]
+    bootstrap_python: Optional[str] = None
+
+    def create_argv(self, pex_path: str, *args: str) -> Iterable[str]:
+        argv = [self.bootstrap_python] if self.bootstrap_python else []
+        argv.extend((pex_path, *args))
+        return argv
+
+    @property
+    def environment_dict(self) -> Mapping[str, str]:
+        return dict(
+            PATH=create_path_env_var(self.path),
+            PEX_PYTHON_PATH=create_path_env_var(self.interpreter_search_paths),
+        )
+
+    def level(self) -> Optional[LogLevel]:
+        return LogLevel.INFO if self.bootstrap_python else LogLevel.WARN
+
+    def message(self) -> Optional[str]:
+        if not self.bootstrap_python:
+            return (
+                "No bootstrap python executable could be found. "
+                "Will attempt to run PEXes directly."
+            )
+        return f"Selected {self.bootstrap_python} to bootstrap PEXes with."
+
+
+class PexRuntimeEnvironment(Subsystem):
+    """How Pants uses Pex to run Python subprocesses."""
+
+    options_scope = "pex"
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--binary-search-path",
+            advanced=True,
+            type=list,
+            default=["<PATH>"],
+            metavar="<binary-paths>",
+            help=(
+                "A list of paths to search for binaries needed by PEX subprocess. The special "
+                'string "<PATH>" will expand to the contents of the PATH env var.'
+            ),
+        )
+        register(
+            "--bootstrap-interpreter-names",
+            advanced=True,
+            type=list,
+            default=["python", "python3", "python2"],
+            metavar="<bootstrap-python-names>",
+            help=(
+                "The names of python binaries to search for to bootstrap PEX files with. Earlier "
+                "names in the list will be preferred over later names and the first matching "
+                "interpreter found will be used to execute the PEX bootstrap stage. If no matching "
+                "interpreters are found, PEXes will be executed directly relying on their embedded "
+                "shebang and the $PATH (see --binary-search-path) to locate a bootstrap "
+                "interpreter."
+            ),
+        )
+
+    @memoized_property
+    def path(self) -> Tuple[str, ...]:
+        def iter_path_entries():
+            for entry in self.options.binary_search_path:
+                if entry == "<PATH>":
+                    # TODO(#9760): This is not very robust. We want to be able to read an env var
```

Comment:
To be clear, rather than exposing an "env" intrinsic, I expect that the rest of #9760 (which I should probably break out) will deprecate the `binary_search_path` option, and move the PATH to being a global option that is only optionally overridden per Process.